app/models/components/course/groups_ability_component.rb

```diff
@@ -22,6 +22,7 @@ def allow_group_manager_manage_group
   end
 
   def course_group_manager_hash
-    { group_users: { user_id: user.id, role: Course::GroupUser.roles[:manager] } }
+    { group_users: { course_user_id: user.course_user_ids,
```

Comment:
I tried that, but I suspect there is some issue with the construction of the query - probably to do with `cancancan-squeel` gem >_< 
```
NoMethodError:
  undefined method `klass' for nil:NilClass
  Did you mean?  class
  # /Users/weiqingtoh/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/cancancan-squeel-0.1.3/lib/cancancan/squeel/expression_builder.rb:68:in `build_association_comparison_node'
```